docs/one_collect-design.md

```diff
@@ -0,0 +1,651 @@
+
+```rust
+use one_collect::helpers::exporting::record::*;
+
+// Universal record types work across Linux and Windows
+let process_record = UniversalProcessRecord {
+    pid: 1234,
+    name: "myapp.exe".to_string(),
+    command_line: "myapp.exe --verbose".to_string(),
+    // ... other fields
+};
+```
+
+##### End-to-End Example
+
+Here's a complete example showing how to capture CPU profiling data and export to multiple formats:
+
+```rust
+use one_collect::helpers::exporting::*;
+use one_collect::helpers::exporting::universal::*;
+use std::time::Duration;
+
+// 1. Configure export settings
+let mut settings = ExportSettings::new();
+settings.set_cpu_sampling_interval(Duration::from_millis(1));
+settings.enable_callstacks(true);
+settings.set_output_path("profile_trace");
+
+// Enable CPU profiling scenario - this automatically:
+// - Determines what events are needed (timer, context switch, etc.)
+// - Handles rundown for existing processes
+// - Sets up appropriate event handlers
+settings.add_scenario("cpu_profiling_with_stacks");
+
+// 2. Create universal exporter
+let mut exporter = UniversalExporter::new(settings);
+
+// 3. Add export hooks for different formats
+exporter.add_export_hook(|machine: &Writable<ExportMachine>| {
+    // Export as PerfView-compatible ETL format
+    machine.read(|export_machine| {
+        export_machine.export_perfview("profile.etl")?;
+        Ok(())
+    })
+});
+
+exporter.add_export_hook(|machine: &Writable<ExportMachine>| {
+    // Export as .NET nettrace format from same data
+    machine.read(|export_machine| {
+        export_machine.export_nettrace("profile.nettrace")?;
+        Ok(())
+    })
+});
+
+// 4. Start collection
+let machine = exporter.start_collection("profiling_session")?;
+
+// 5. Collection runs automatically...
+std::thread::sleep(Duration::from_secs(10));
+
+// 6. Stop and export
+exporter.stop_and_export()?;
+```
+
+##### Scenario-Based Model
+
+The Universal Export Framework moves beyond individual event specification to a scenario-based approach:
+
+```rust
+// Instead of manually specifying events:
+// session.add_event(timer_event);
+// session.add_event(context_switch_event);
+// session.add_event(process_event);
+// ... dozens more
+
+// Simply enable scenarios:
+settings.add_scenario("cpu_profiling");           // CPU sampling + stacks
+settings.add_scenario("memory_tracking");         // Heap allocations
+settings.add_scenario("io_monitoring");           // File/network I/O
+settings.add_scenario("dotnet_gc_analysis");      // .NET GC events
+
+// Each scenario automatically determines required events
+// and handles cross-platform differences
+```
+
+#### Helpers System (`helpers` module)
+
+##### Callstack Helper (`helpers::callstack`)
+Stack unwinding integration:
+- Integrates with `ruwind` library
+- Manages unwinding context and state
+- Provides callstack symbolization
+
+##### .NET Helper (`helpers::dotnet`)
+Managed code profiling support:
```

Comment:
This is incorrect.  The dotnet helper:

- Enables jitted code symbol capture.
- Enables .NET event capture control (enable/disable).